File: Vectors , ArrayList, List/Java/PairSum.java
// Return the indexes of elements whose sum is equal to the target.
import java.util.ArrayList;
import java.util.Arrays;

public class PairSum {
    public static ArrayList<Integer> pairSum(ArrayList<Integer> v, int target) {
        ArrayList<Integer> ans = new ArrayList<>();
        int start = 0;
        int end = v.size() - 1;
        while (start < end) {
            if (v.get(start) + v.get(end) == target) {
                ans.add(start);
                ans.add(end);
                return ans;
            } else if (v.get(start) + v.get(end) < target) {
                start++;
            } else {
                end--;
            }
        }
        return ans;
    }

    public static void main(String[] args) {
        ArrayList<Integer> v = new ArrayList<>(Arrays.asList(2, 7, 11, 15));
        int target = 9;
        ArrayList<Integer> ans = pairSum(v, target);
        System.out.println(ans.get(0) + ", " + ans.get(1));
    }
}

File: Vectors , ArrayList, List/C++/PairSum.cpp
// Return the indexes of elements whose sum is equal to the target.

#include <iostream>
#include <vector>

using namespace std;

vector<int> pairSum(vector<int> v, int target){
    vector<int> ans;
    int start = 0;
    int end = v.size() - 1;
    while (start < end){
        if (v[start] + v[end] == target){
            ans.push_back(start);
            ans.push_back(end);
            return ans;
        }
        else if (v[start] + v[end] < target){
            start++;
        }
        else{
            end--;
        }
    }
    return ans;
}

int main(){
    vector<int> v = {2, 7, 11, 15};
    int target = 9;
    vector<int> ans = pairSum(v, target);
    cout << ans[0] << ", " << ans[1] << endl;
    return 0;
}